packages/esm-patient-chart-app/src/encounter-list/components/encounter-list.component.tsx

```diff
@@ -0,0 +1,300 @@
+import { navigate, showModal, showSnackbar } from '@openmrs/esm-framework';
+import { EmptyState } from '@openmrs/esm-patient-common-lib';
+import { useTranslation } from 'react-i18next';
+import { EncounterListDataTable } from './table.component';
+import {
+  Button,
+  Link,
+  OverflowMenu,
+  OverflowMenuItem,
+  DataTableSkeleton,
+  MenuButton,
+  MenuItem,
+  Pagination,
+} from '@carbon/react';
+import { Add } from '@carbon/react/icons';
+import { launchEncounterForm } from '../../clinical-views/utils/helpers';
+import { deleteEncounter } from '../encounter-list.resource';
+import { useEncounterRows, useFormsJson, usePatientDeathStatus } from '../hooks';
+
+import styles from './encounter-list.scss';
+
+export interface EncounterListColumn {
+  key: string;
+  header: string;
+  getValue: (encounter: any) => string;
+  link?: any;
+}
+
+export interface EncounterListProps {
+  patientUuid: string;
+  encounterType: string;
+  columns: Array<any>;
+  headerTitle: string;
+  description: string;
+  formList?: Array<{
+    name?: string;
+    uuid: string;
+    excludedIntents?: Array<string>;
+    fixedIntent?: string;
+    isDefault?: boolean;
+  }>;
+  launchOptions: {
+    hideFormLauncher?: boolean;
+    displayText?: string;
+    workspaceWindowSize?: 'minimized' | 'maximized';
+  };
+  filter?: (encounter: any) => boolean;
+  afterFormSaveAction?: () => void;
+}
+
+export const EncounterList: React.FC<EncounterListProps> = ({
+  patientUuid,
+  encounterType,
+  columns,
+  headerTitle,
+  description,
+  formList,
+  filter,
+  launchOptions,
+  afterFormSaveAction,
+}) => {
+  const { t } = useTranslation();
+  const { isDead } = usePatientDeathStatus(patientUuid);
+  const formUuids = useMemo(() => formList.map((form) => form.uuid), [formList]);
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pageSize, setPageSize] = useState(10);
+
+  const { formsJson, isLoading: isLoadingFormsJson } = useFormsJson(formUuids);
+
+  const { encounters, total, isLoading, onFormSave, mutate } = useEncounterRows(
+    patientUuid,
+    encounterType,
+    filter,
+    afterFormSaveAction,
+    pageSize,
+    currentPage,
+  );
+
+  const { workspaceWindowSize, displayText, hideFormLauncher } = launchOptions;
+
+  const defaultActions = useMemo(
+    () => [
+      {
+        label: t('viewEncounter', 'View'),
+        form: {
+          name: formsJson[0]?.name,
+        },
+        mode: 'view',
+        intent: '*',
+      },
+      {
+        label: t('editEncounter', 'Edit'),
+        form: {
+          name: formsJson[0]?.name,
+        },
+        mode: 'edit',
+        intent: '*',
+      },
+      {
+        label: t('deleteEncounter', 'Delete'),
+        form: {
+          name: formsJson[0]?.name,
+        },
+        mode: 'delete',
+        intent: '*',
+      },
+    ],
+    [formsJson, t],
+  );
+
+  const createLaunchFormAction = (encounter, mode) => () =>
```

Comment:
This should be wrapped in `useCallback()`. Also, `encounter`, and `mode` should be typed.